File: src/main/java/Input.java
import org.lwjgl.glfw.GLFWKeyCallback;
import static org.lwjgl.glfw.GLFW.*;


public class Input extends GLFWKeyCallback{

    public static boolean[] keys = new boolean[65535];

    @Override
    public void invoke(long window, int key, int scancode, int action, int mods) {
        keys[key] = action != GLFW_RELEASE;
    }

    //LUKA, DONT FORGET THIS!

    //inside main class:
    //  private GLFWKeyCallback keyCallback; (attribute)

    //inside main init() method:
    //  glfwSetKeyCallback(window, keyCallback = new Input());

    //inside main update() method:
    //  glfwPollEvents();
    //  if(Input.keys[GLFW_KEY_SPACE])
    //      {whatever happens after pressing SPACE}

    //at the end of main run() method:
    //  keyCallback.release();








}

File: src/main/java/Input.kt
import org.lwjgl.glfw.GLFW
import org.lwjgl.glfw.GLFW.GLFW_RELEASE
import org.lwjgl.glfw.GLFWKeyCallback


class Input : GLFWKeyCallback() {

    val keys: Array<Boolean> = Array(65535) { false }

    override operator fun invoke(window: Long, key: Int, scancode: Int, action: Int, mods: Int) {
        keys[key] = action != GLFW_RELEASE
        if (keys[key])
            println(key)
        handleEscape(key, action)
    }

    private fun handleEscape(key: Int, action: Int) {
        if (key == GLFW.GLFW_KEY_ESCAPE && action == GLFW_RELEASE)
            Window.setWindowShouldClose()
    }

}
